### PR TITLE
feat: verbose message for toFinishWith failed assertions

### PR DESCRIPTION
**Pull request by ruocco-l**
Reworks `toFinishWith` to push verbose explicative messages on what actually failed.

@kazadoiyul

### CODE REVIEW

File: lib/extend-expect.ts [hunk 1]
```diff
@@ -193,12 +211,13 @@ export const extendExpect = (expect: ExpectStatic): ExpectStatic => {
...
                     diffs.actual.push(` logs=[${occuredLogs.join(', ')}]`);
                     diffs.expected.push(` logs=[]`);
+                    failedAssertions.push(`Failed forbidden logs check.`);
                 }
             }
 
             return {
                 pass: diffs.pass,
-                message: () => `Run ${received.id} didn't finish as expected`,
+                message: () => `Run did not finish as expected. Failed assertions: ${failedAssertions.join(' ',)}`,
```
Comment (kazadoiyul) [inline lib/extend-expect.ts:222]: why did you remove runId from the log? is it present in failed assertions or is it just a bug?

**Review by kazadoiyul — APPROVED** [review]
i like more detailed log, thanks a lot!

File: lib/extend-expect.ts (hunk 1 above)
Comment (ruocco-l) [reply lib/extend-expect.ts:222]: Yeah, that's because in the last PR we now get links to every run along with the failing actor name. Logs will look like this
```
AssertionError: Fake test to show input values in the report: expected false to be true // Object.is equality --- <https://console.apify.com/view/runs/pgP2uc6eKQgD6H1J7|streamers/youtube-channel-scraper>
        blocks: • Error: Run did not finish as expected. Failed assertions: Failed duration check. --- <https://console.apify.com/view/runs/pgP2uc6eKQgD6H1J7|streamers/youtube-channel-scraper>
                • AssertionError: Fake test to show input values in the report: expected false to be true // Object.is equality --- <https://console.apify.com/view/runs/Jwx49cEPs5fuoSg9d|streamers/youtube-comments-scraper>

```

**Review by foxt451 — COMMENTED** [review]
Maybe not related to this PR, but could you also add logic to show the full diff from the underlying test lib? Like you can get `Error: Expected "undefined" to be a non-empty string` and have no idea what it means unless you navigate to the job

File: lib/extend-expect.ts [hunk 2]
```diff
@@ -125,9 +127,13 @@ export const extendExpect = (expect: ExpectStatic): ExpectStatic => {
...
                 const actual = received.status;
-                diffs.pass = expected === actual;
-                diffs.actual.push(`status=${actual}`);
-                diffs.expected.push(`status=${expected}`);
+                const statusTest = expected === actual;
+                if (statusTest === false) {
+                    diffs.pass = false;
+                    diffs.actual.push(`status=${actual}`);
+                    diffs.expected.push(`status=${expected}`);
+                    failedAssertions.push(`Failed status check.`);
```
Comment (oklinov) [inline lib/extend-expect.ts]: optional:
```suggestion
                    failedAssertions.push(`Failed status check, expected "${expected}", got "${actual}".`);
```

we'd then get:
```
Run did not finish as expected. Failed assertions: Failed status check, expected "TIMING-OUT", got "SUCCEEDED".
```

**Review by oklinov — APPROVED** [review]
LGTM..I just left one optional suggestion...thanks and sorry for the late review

**Comment by ruocco-l** [comment]
I added the `expected` vs `actual` in the failed assertion message. If there are multiple assertions failing (e.g. `datasetCount` and `ppeEventCount`, see screenshot) they are written in the same line and it's not the best, but adding other character or line breaks can be also confusing (and potentially tricky, since
 we are counting on the line break to create the status message). Maybe we can wait and see how they look on Slack when we have a run failing big.

Screenshot 2026-01-08 at 13 00 32

**Review by oklinov — APPROVED** [review]
